### PR TITLE
records: adds new CMS ML software records

### LINKED ISSUE #2584: CMS: SW records for ML sample production (Run2 QCD for jettuples)
From #2447 
The structure can be as in http://opendata-dev.cern.ch/record/5104

@caredg @laramaktub This is what we would need to extract from a github repo or CAP in order to build a SW record on the portal

Needed:
- title: JetNtupleProducerTool - a Jet tuple producer from CMS Run2 MiniAOD
- authors: Kimmo Kallonen @kimmokal others?
- tags/keywords: Software, Tool
- the datasets that are used as input: recid 12021 (or any Run2 MiniAOD containing jets)
- System details 
   - VM recid 252
   - CMSSW version: CMSSW_8_0_26
- how can you use, the github repo is in 
https://github.com/cms-legacydata-analyses/JetNtupleProducerTool/tree/2016
but it will be a placholder as it will be moved to https://github.com/cms-opendata-analyses/JetNtupleProducerTool/tree/2016 and have a release tag there
- licence @kimmokal do you agree with GNU General Public License (GPL) version 3

@ArtemisLav @tiborsimko  :
- for run periods, I think we really must add Run2, Phase2 instead of specific run periods. It does not matter if they show separately. For Phase2 (=upgrade) SW (another record), we have not yet done the data-taking so it would be impossible to give the exact run periods...
- these ML SW records produce a dataset which is on the portal, for this recid 12100
   - do we have an appropriate json field for that?
   - it should have as description "The ouput dataset of this software tool is available in:"

json could be:

```
{
  "created": "FIXME", 
  "id": FIXME, 
  "metadata": {
    "$schema": "http://opendata.cern.ch/schema/records/record-v1.0.0.json", 
    "abstract": {
      "description": " <p>This is a CMSSW module for producing flat tuples containing jet properties from 13 TeV Run2 MC samples. The code is intended to run inside the CMS Open Data environment</p>"
    }, 
    "accelerator": "CERN-LHC", 
    "authors": [
      {
        "name": "Kallonen, Kimmo"
      }
    ], 
    "collections": [
      "FIXME Is this needed/necessary: CMS-Validation-Utilities"
    ], 
    "control_number": "FIXME", 
    "date_created": [
      "2019"
    ], 
    "date_published": "2019", 
    "distribution": {
      "formats": [
        "gz"
      ], 
      "number_files": 1, 
      "size": FIXME
    }, 
    "doi": "FIXME", 
    "experiment": "CMS", 
    "files": [
      {
        "checksum": "FIXME", 
        "filename": "FIXME check JetNtupleProducerTool-1.0.0.tar.gz", 
        "size": 16966, 
        "uri_http": "FIXME", 
        "uri_root": "FIXME"
      }
    ], 
    "index_files": [], 
    "license": {
      "attribution": "GNU General Public License (GPL) version 3"
    }, 
    "publisher": "CERN Open Data Portal", 
    "recid": "FIXME", 
    "run_period": [
      "Run2"
    ], 
    "system_details": {
      "description": "Use this code with the CMS Open Data VM environment", 
      "recid": "252", 
      "release": "CMSSW_8_0_26"
    }, 
    "title": "JetNtupleProducerTool - a Jet tuple producer from CMS Run2 MiniAOD", 
    "type": {
      "primary": "Software", 
      "secondary": [
        "Tool"
      ]
    }, 
    "usage": {
      "description": " <p>If you do not have the CERN Virtual Machine for CMS open data installed, follow the instructions in step 1 at <a href=\"/VM/CMS/2011\">How to install a CERN Virtual Machine</a>.  <p>To run the analysis, follow the instructions in FIXME <a href=\"https://github.com/cms-legacydata-analyses/JetNtupleProducerTool/tree/2016\">https://github.com/cms-legacydata-analyses/JetNtupleProducerTool/tree/2016</a>.</p> "
    }, 
    "use_with": {
      "description": "Use this with Run2 QCD MiniAODSIM dataset (or any Run2 MiniAOD containing jets).", 
      "links": [
        {
          "recid": "12021"
        }
      ]
    }
  }, 
  "updated": "FIXME"
}
```

### PR DESCRIPTION
**Pull request by ArtemisLav**
* (closes #2584)

Signed-off-by: Artemis Lavasa <artemis.lavasa@cern.ch>

### CODE REVIEW

**Comment by ArtemisLav** [comment]
Initial draft - many FIXME's to work through.

**Comment by ArtemisLav** [comment]
@tiborsimko does it make sense to take care of https://github.com/cernopendata/opendata.cern.ch/issues/2585 in this PR as well (i.e. add to this file)?

File: cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    ],
+    "date_published": "2019",
+    "distribution": {
+      "formats": [
+        "gz"
+      ],
+      "number_files": 1,
+      "size": 11111111
+    },
+    "doi": "FIXME",
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: We can only mint DOI once the example will be having some GitHub tag.  Note that they may move to a different GitHub organisation... so I'll merge without DOI parts for now.

File: cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "experiment": "CMS",
+    "files": [
+      {
+      }
+    ],
+    "license": {
+      "attribution": "GNU General Public License (GPL) version 3"
+    },
+    "publisher": "CERN Open Data Portal",
+    "recid": "FIXME",
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: 12101 for example, we can start incrementing there for all datascience material, datasets and samples (until ~13000)

File: cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    "accelerator": "CERN-LHC",
+    "authors": [
+      {
+        "name": "Kallonen, Kimmo",
+        "orcid": "0000-0001-9769-7163"
+      }
+    ],
+    "collections": [
+      "CMS-Validation-Utilities"
```
Comment (tiborsimko) [inline cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: Hmm, this is not validation code, although I see we used it in the past... http://opendata-dev.cern.ch/record/5104

Comment (tiborsimko) [reply cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: ... should be rather put under `CMS-Tools` collection

Comment (katilp) [reply cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: OK for `CMS-Tools` (or leave out)

**Comment by katilp** [comment]
As discussed with @tiborsimko , the file can be renamed to cms-tools-Run2-datascience.json and it can host #2585 (I will make a separate PR for that)

File: cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json (hunk 3 above)
Comment (tiborsimko) [reply cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: Amended

File: cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json (hunk 2 above)
Comment (tiborsimko) [reply cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: Amended

File: cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json (hunk 1 above)
Comment (tiborsimko) [reply cernopendata/modules/fixtures/data/records/cms-SW-FIXMEfilename.json]: Amended

**Pull request merged**